api/swagger.yml

```diff
@@ -5582,6 +5653,159 @@ paths:
+            description: filter pull requests by state
+      responses:
+        200:
+          description: list of pull requests
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PullRequestsList"
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        404:
+          $ref: "#/components/responses/NotFound"
+        420:
+          description: too many requests
+        default:
+          $ref: "#/components/responses/ServerError"
+    post:
+      tags:
+        - pulls
+        - experimental
+      operationId: createPullRequest
+      summary: create pull request
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/PullRequestCreation"
+      responses:
+        201:
+          description: pull request id
+          content:
+            text/html:
+              schema:
+                type: string
+        400:
+          $ref: "#/components/responses/ValidationError"
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        403:
+          $ref: "#/components/responses/Forbidden"
+        404:
+          $ref: "#/components/responses/NotFound"
+        409:
+          $ref: "#/components/responses/Conflict"
+        420:
+          description: too many requests
+        default:
+          $ref: "#/components/responses/ServerError"
+
+  /repositories/{repository}/pulls/{pull_request}:
+    parameters:
+      - in: path
+        name: repository
+        required: true
+        schema:
+          type: string
+      - in: path
+        name: pull_request
+        required: true
+        description: pull request id
+        schema:
+          type: string
+    get:
+      tags:
+        - pulls
+        - experimental
+      operationId: getPullRequest
+      summary: get pull request
+      responses:
+        200:
+          description: pull request
+          content:
+            application/json:
+              schema:
+                $ref: "#/components/schemas/PullRequest"
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        404:
+          $ref: "#/components/responses/NotFound"
+        420:
+          description: too many requests
+        default:
+          $ref: "#/components/responses/ServerError"
+    delete:
+      tags:
+        - pulls
+        - experimental
+      operationId: deletePullRequest
+      summary: delete pull request
+      responses:
+        204:
+          description: pull request deleted successfully
+        401:
+          $ref: "#/components/responses/Unauthorized"
+        403:
+          $ref: "#/components/responses/Forbidden"
+        404:
+          $ref: "#/components/responses/NotFound"
+        420:
+          description: too many requests
+        default:
+          $ref: "#/components/responses/ServerError"
+    patch:
```

Comment:
Note that in the Design [Issue](https://github.com/treeverse/lakeFS/issues/8164) it's designed different - 
But what's written in the PR is far better (hence I think the design should be updated).